### PR TITLE
 Add convolution unit test for Arm backend

### PR DESCRIPTION
**Pull request by freddan80**
* Added both depthwise and regular conv
* Removed corresponding conv legacy unit tests
* Fix zero point usage solves issues with random input
* Removed add/add2 legacy unit tests as they are already implemented
* Bump serialization lib submodule to avoid warning spam
* Fixed rounding issue tosa_test_utils

### CODE REVIEW

**Comment by freddan80** [comment]
@digantdesai  Rebase conflict expected with PR #2371 , so hold off merging this PR until that one has gone in. I'll rebase as soon as it's in.

**Comment by freddan80** [comment]
@digantdesai CI failure not related to this change.

File: backends/arm/test/models/test_mobilenet_v2_arm.py [hunk 1]
```diff
@@ -6,26 +6,22 @@
...
-# Running the jobs in _unittest.yml will not work since that environment don't
-# have the vela tool, nor the tosa_reference_model tool. Hence, we need a way to
-# run what we can in that env temporarily. Long term, vela and tosa_reference_model
-# should be installed in the CI env.
-TOSA_REF_MODEL_INSTALLED = shutil.which("tosa_reference_model")
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
+torch.manual_seed(42)
```
Comment (digantdesai) [inline backends/arm/test/models/test_mobilenet_v2_arm.py:23]: remove this?

File: backends/arm/test/ops/test_add.py [hunk 2]
```diff
@@ -1,32 +1,26 @@
...
-# have the vela tool, nor the tosa_reference_model tool. Hence, we need a way to
-# run what we can in that env temporarily. Long term, vela and tosa_reference_model
-# should be installed in the CI env.
-TOSA_REF_MODEL_INSTALLED = shutil.which("tosa_reference_model")
-VELA_INSTALLED = shutil.which("vela")
-
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
+torch.manual_seed(42)
```
Comment (digantdesai) [inline backends/arm/test/ops/test_add.py:22]: this too?

File: backends/arm/test/ops/test_conv.py [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+testsuite = [
+    ("2x2_3x1x40x40_nobias", conv2d_2x2_3x1x40x40_nobias),
+    ("3x3_1x3x256x256_st1", conv2d_3x3_1x3x256x256_st1),
+    ("3x3_1x3x12x12_st2_pd1", conv2d_3x3_1x3x12x12_st2_pd1),
+    ("1x1_1x2x128x128_st1", conv2d_1x1_1x2x128x128_st1),
+    ("2x2_1x1x14x14_st2", conv2d_2x2_1x1x14x14_st2),
+    ("5x5_3x2x128x128_st1", conv2d_5x5_3x2x128x128_st1),
+    ("3x3_1x3x224x224_st2_pd1", conv2d_3x3_1x3x224x224_st2_pd1),
+    ("two_conv2d_nobias", two_conv2d_nobias),
+    ("two_conv2d", two_conv2d),
```
Comment (digantdesai) [inline backends/arm/test/ops/test_conv.py:234]: Do we also want to add memory format tests? i.e. if we can correctly handle `nchw` and `nchw` or not through conversions. cc @Jerry-Ge

File: backends/arm/test/ops/test_depthwise_conv.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+        tester = (
+            ArmTester(
+                module,
+                inputs=test_data,
+                profile=TosaProfile.MI,
+                backend=ArmBackendSelector.TOSA,
+                permute_memory_to_nhwc=True,
+            )
+            .export()
+            .to_edge()
```
Comment (digantdesai) [inline backends/arm/test/ops/test_depthwise_conv.py:144]: check for conv nodes?

File: backends/arm/test/ops/test_conv.py [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+        tester = (
+            ArmTester(
+                module,
+                inputs=test_data,
+                profile=TosaProfile.MI,
+                backend=ArmBackendSelector.TOSA,
+                permute_memory_to_nhwc=True,
+            )
+            .export()
+            .to_edge()
```
Comment (digantdesai) [inline backends/arm/test/ops/test_conv.py:258]: check for conv nodes

File: backends/arm/test/ops/test_conv.py [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+            .partition()
+            .check_count({"torch.ops.higher_order.executorch_call_delegate": 1})
```
Comment (digantdesai) [inline backends/arm/test/ops/test_conv.py:260]: `check_not` for conv nodes?

File: backends/arm/third-party/serialization_lib
Comment (digantdesai) [inline backends/arm/third-party/serialization_lib:1]: I see multiple PRs updating this, hopefully tools can handle it

**Review by digantdesai — APPROVED** [review]

File: backends/arm/test/ops/test_conv.py (hunk 3 above)
Comment (Jerry-Ge) [reply backends/arm/test/ops/test_conv.py:234]: question: all those conv2d tests won't handle `nchw` in TOSA, so what's the purpose of adding this test?

File: backends/arm/test/ops/test_add.py (hunk 2 above)
Comment (freddan80) [reply backends/arm/test/ops/test_add.py:22]: `torch.randn` It's used in this file.

File: backends/arm/test/ops/test_conv.py (hunk 3 above)
Comment (freddan80) [reply backends/arm/test/ops/test_conv.py:234]: True, it won't be handled in TOSA. This PR is mainly porting unit tests to the new test structure (and adding some test width).

File: backends/arm/test/ops/test_depthwise_conv.py (hunk 4 above)
Comment (freddan80) [reply backends/arm/test/ops/test_depthwise_conv.py:144]: I thought about doing that, but it got a bit convoluted with the multi-conv-op tests. That and the fact that the main purpose of the test is to ensure the stages from partition and onwards are what is unique to this Arm backend unit test, made me go without that check. I can add it if you have a string opinion about that. Let me know.

File: backends/arm/test/ops/test_conv.py (hunk 5 above)
Comment (freddan80) [reply backends/arm/test/ops/test_conv.py:258]: dito

File: backends/arm/third-party/serialization_lib
Comment (freddan80) [reply backends/arm/third-party/serialization_lib:1]: Should be fine I think since they're identical.

**Comment by freddan80** [comment]
CI fail unrelated to this PR. Credential issues:

```
ExecuTorch: Failed to run runner.
ERROR conda.cli.main_run:execute(47): `conda run bash backends/apple/coreml/scripts/build_all.sh` failed. (See above for error)
    main()
  File "/Users/runner/work/executorch/executorch/test-infra/.github/scripts/run_with_env_secrets.py", line 60, in main
    run_cmd_or_die(f"bash { os.environ.get('RUNNER_TEMP', '') }/exec_script")
  File "/Users/runner/work/executorch/executorch/test-infra/.github/scripts/run_with_env_secrets.py", line 38, in run_cmd_or_die
    raise RuntimeError(f"Command {cmd} failed with exit code {exit_code}")
RuntimeError: Command bash /Users/runner/work/_temp/exec_script failed with exit code 1
Error: Process completed with exit code 1.
```

File: backends/arm/test/ops/test_add.py (hunk 2 above)
Comment (digantdesai) [reply backends/arm/test/ops/test_add.py:22]: I guess it is ok to run test each time with a new random value? You can set the seed for debugging locally.

File: backends/arm/test/ops/test_conv.py (hunk 3 above)
Comment (digantdesai) [reply backends/arm/test/ops/test_conv.py:234]: OK

File: backends/arm/test/ops/test_add.py (hunk 2 above)
Comment (freddan80) [reply backends/arm/test/ops/test_add.py:22]: I see your point. Let us address that in coming PRs.

**Pull request closed without merging**